tomb/src/pipelines/banyan_api/bucket/keys.rs

```diff
@@ -0,0 +1,102 @@
+use tomb_common::{
+    banyan_api::{client::Client, error::ClientError, models::bucket_key::BucketKey},
+    metadata::FsMetadata,
+};
+use tomb_crypt::prelude::{PrivateKey, PublicKey};
+use uuid::Uuid;
+
+use crate::{
+    cli::command::{KeySpecifier, KeySubCommand},
+    pipelines::error::TombError,
+    types::config::globalconfig::GlobalConfig,
+};
+
+/// Pipeline for processing BucketKey commands
+pub(crate) async fn pipeline(
+    global: &GlobalConfig,
+    client: &mut Client,
+    command: KeySubCommand,
+) -> Result<String, TombError> {
+    match command {
+        // List Keys
+        KeySubCommand::List(bucket_specifier) => {
+            BucketKey::read_all(global.get_bucket_id(&bucket_specifier)?, client)
+                .await
+                .map(|keys| {
+                    keys.iter()
+                        .fold(String::new(), |acc, key| format!("{}\n\n{}", acc, key))
+                })
+                .map_err(TombError::client_error)
+        }
+        // Create a new key
+        KeySubCommand::Create(bucket_specifier) => {
+            let private_key = &global.wrapping_key().await?;
+            let public_key = private_key
+                .public_key()
+                .map_err(ClientError::crypto_error)?;
+            // Compute PEM
+            let pem = String::from_utf8(
+                public_key
+                    .export()
+                    .await
+                    .map_err(ClientError::crypto_error)?,
+            )
+            .unwrap();
+
+            // Get Bucket
+            let bucket = global.get_bucket_by_specifier(&bucket_specifier)?;
+            let mut fs = FsMetadata::unlock(private_key, &bucket.metadata).await?;
+            fs.share_with(&public_key, &bucket.metadata).await?;
+            fs.save(&bucket.metadata, &bucket.metadata).await?;
+
+            if let Some(remote_id) = bucket.remote_id {
+                BucketKey::create(remote_id, pem, client)
+                    .await
+                    .map(|key| format!("{}", key))
+                    .map_err(TombError::client_error)
+            } else {
+                Ok("added key to bucket locally".to_string())
+            }
+        }
```

Comment:
I'm not sure when this creation would be useful and I'm guessing this was what you were asking about before. If this is the approval side of the public key it shouldn't be generating a key, it should be using the public portion of a key already present as a key request on the metadata server.

I don't otherwise see an approve action here or something similar so I'm guessing this is the approval command.

Reply:
Ahhh yeah, that's the request access workflow. So maybe adjust the name to something along the lines of request access to be a bit more clear